src/main/java/com/sparta/popupstore/domain/promotionevent/scheduler/PromotionEventEndCouponDeleteScheduler.java

```diff
@@ -1,38 +1,51 @@
 package com.sparta.popupstore.domain.promotionevent.scheduler;
 
+import com.sparta.popupstore.domain.common.util.ValidUtil;
+import com.sparta.popupstore.domain.promotionevent.entity.PromotionEvent;
 import com.sparta.popupstore.domain.promotionevent.repository.CouponRepository;
 import com.sparta.popupstore.domain.promotionevent.repository.PromotionEventRepository;
+import com.sparta.popupstore.s3.S3ImageService;
 import jakarta.transaction.Transactional;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
+import java.time.LocalDateTime;
+import java.util.List;
+
 @Slf4j
 @Component
 @RequiredArgsConstructor
 public class PromotionEventEndCouponDeleteScheduler {
     private final PromotionEventRepository promotionEventRepository;
     private final CouponRepository couponRepository;
+    private final S3ImageService s3ImageService;
 
     @Transactional
     @Scheduled(fixedRate = 300000) // 테스트 용
-    @Scheduled(cron = "00 1 0 * * *") // 매일 00시 1분
+    @Scheduled(cron = "0 0 0 * * *") // 매일 00시 00분
     public void hardDeletePromotionEvent() {
         log.info("hardDeletePromotionEvent 스케줄러");
+        List<PromotionEvent> eventList = promotionEventRepository.findAllByEndDateTimeAfterSixMonths();
+        for(PromotionEvent event : eventList){
+            if(ValidUtil.isValidNullAndEmpty(event.getImageUrl())){
+                s3ImageService.deleteImage(event.getImageUrl());
+            }
+        }
         promotionEventRepository.deletePromotionEventByDeletedAtAfterSixMonths();
```

Comment:
```suggestion
        promotionEventRepository.deleteAll(eventList);
```
삭제할 리스트를 위에서 찾았으니 활용하면 좋을 것 같습니다.
아주 약간의 시간 차이로 위에서 이미지를 삭제한 이벤트와 여기서 삭제하는 이벤트가 다를 수도 있어서요.